todo/src/App.js

```diff
@@ -0,0 +1,60 @@
+import React, { Component } from "react";
+import "./App.css";
+import { connect } from "react-redux";
+import { addTodo } from "./actions";
+import TodoList from "./components/TodoList";
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      newTodo: ""
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.updateNewTodo = this.updateNewTodo.bind(this);
+  }
+
+  addTodo(event) {
+    event.preventDefault();
+    this.props.addTodo({
+      value: this.state.newTodo,
+      complete: false
+    });
+    this.setState({
+      newTodo: ""
+    });
+  }
+
+  updateNewTodo(event) {
+    this.setState({
+      newTodo: event.target.value
+    });
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <h1>To Do List</h1>
+        <form onSubmit={this.addTodo}>
+          <input
+            onChange={this.updateNewTodo}
+            placeholder="New To Do"
+            value={this.state.newTodo}
+          />
+        </form>
+        <TodoList todos={this.props.todos} />
+      </div>
+    );
+  }
+}
+
+const mapStateToProps = state => {
+  return {
+    todos: state.todos
+  };
+};
+
+export default connect(
+  mapStateToProps,
+  { addTodo }
```

Comment:
```suggestion
  { addTodo, toggleComplete }
```
If you did this, you could pass toggleComplete down to the todo component without having to connect the todo component. Don't forget to import it at the top if you choose this route